### PR TITLE
Make pysteps optional

### PR DESCRIPTION
**Pull request by lucyleeow**
Addresses #1384 
Still need to decide where to put optional dependencies if we remove pysteps from `environment.yml`

Description

Testing:
 - [x] Ran tests and they passed OK
 - [N/A ] Added new tests for the new feature(s)

CLA
 - [x] If a new developer, signed up to CLA

### CODE REVIEW

**Review by tjtg — CHANGES_REQUESTED** [review]
Thanks for the contribution - all the code changes look OK.
As this is your first pull request, please agree to the CLA and add yourself to the list of contributors.

**Review by tjtg — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -61,6 +61,7 @@ below:
 * Eleanor Smith (Met Office, UK)
 * Tomasz Trzeciak (Met Office, UK)
 * Mark Worsfold (Met Office, UK)
+* Lucy Liu (Bureau of Meteorology, Australia)
```
Comment (gavinevans) [inline CONTRIBUTING.md:64]: Just a quick comment, could this be added in alphabetical order? (The list is currently in alphabetical order by surname.)

Comment (lucyleeow) [reply CONTRIBUTING.md]: Thanks - only checked the first names for alphabetical order!

**Comment by lucyleeow** [comment]
Should pysteps be removed from `environment.yml` and put in a different list of optional dependencies...?

**Comment by tjtg** [comment]
> Should pysteps be removed from `environment.yml` and put in a different list of optional dependencies...?

Generally, it'd be useful to clearly list what is required and what is optional (eg. only required for particular plugins within in IMPROVER). However, we need to make sure that at least one of the environments used for the CI tests (#1381 and #1392) continues to include everything so all the code gets covered by the test run.

**Review by tjtg — DISMISSED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (gavinevans) [reply CONTRIBUTING.md]: No worries 👍

**Comment by cgsandford** [comment]
We are only using the `pysteps.extrapolate` subpackage - we are not using any of the aspects that require open CV etc.  Is there a way to specify dependencies for just that?

**Comment by tjtg** [comment]
@cgsandford 
> We are only using the `pysteps.extrapolate` subpackage - we are not using any of the aspects that require open CV etc. Is there a way to specify dependencies for just that?

Pysteps lists OpenCV as a required dependency rather than an optional one in their [docs](https://github.com/pySTEPS/pysteps/blob/master/doc/source/user_guide/install_pysteps.rst). The [conda-forge recipe](https://github.com/conda-forge/pysteps-feedstock/blob/cd1e10ec30fd5977e402754acdeeccd9c37eafde/recipe/meta.yaml#L30) for pysteps matches with that list in the documentation. Pysteps requirements listed in [setup.py](https://github.com/pySTEPS/pysteps/blob/e2c3484ef234e3ef1ac8f1f11b1e64bc6911f35c/setup.py#L63) are a lot more minimal - this looks to be the advanced user option mentioned in https://github.com/pySTEPS/pysteps/pull/25#issuecomment-445809803

An option would be to list pysteps as a pip dependency rather than conda, but that would have other concerns, hence the preference for this approach:
- More mixing of conda and pip packages, leading to use of multiple package tools and a messier environment. Currently a few packages (clize and its dependencies, timezonefinder) come from pip because they're not available from conda/conda-forge.
- Pysteps packages on pypi are source zips, so the installation is slower due to pip needing to build the package then install it, rather than use a pre-built .whl package.

**Review by tjtg — APPROVED** [review]

**Pull request merged**